### DISCUSSION TITLE
Exploring the Coexistence of Istio and the Gateway API

### DISCUSSION THREAD

**Discussion by seifrajhi** [istio/istio#49971] [Q&A] [answered]
Certainly! Here's the revised full text with a suitable title:

**Title: Exploring the Coexistence of Istio and the Gateway API**

Hello everyone,

After using Istio extensively in our production environment, and we're pretty comfortable with it in our live setups. But we're also keen on staying updated with Kubernetes improvements, particularly  the Gateway API as the successor to legacy ingress solutions.

As per k8s the documentation, it's clear that `Ingress is frozen. New features are being added to the Gateway API` This has led us to consider how Istio and the Gateway API can work together in our AWS EKS clusters, despite their overlapping functionalities and similarities.

And as per Istio docs, while the Gateway API offers a rich set of routing functionalities, it's important to note that it doesn't cover 100% of Istio's feature set. 
However, ongoing efforts are being made, as per the Gateway Enhancement Proposal (GEP), to address these gaps and make the Gateway API more comprehensive.

Therefore, we're seeking guidance on how Istio and the Gateway API can collaborate effectively. Are there any best practices or resources we should be aware of? Any insights or pointers on achieving smooth integration between Istio and the Gateway API would be greatly appreciated!

Can they co-exist or we should choose one.

Thank you in advance for your assistance.

**Comment by howardjohn** [comment, marked as answer]
Currently Istio offers both. You could use both together in one cluster, though it would likely cause confusion (for humans -- the code handles it fine). If you do mix, they need to be against different deployments, though.

For new users, if the feature set of GW API is sufficient I would recommend that. But Istio APIs will stay around as well

**Reply by seifrajhi** [reply]
Thank you @howardjohn, that's clear.

We are big fans of Istio and still consider it a long-term feature in our production setup. However, it is beneficial to explore improvements wherever possible.
We hope that the Istio project will exploit the full potential of the API Gateway and integrate its functionalities to make it more modern and flexible.
